autoendpoint/src/extractors/notification_headers.rs

```diff
@@ -96,6 +94,9 @@ impl NotificationHeaders {
         // Validate encryption if there is a message body
         if has_data {
             headers.validate_encryption()?;
+        } else {
```

Comment:
I'm wondering if we should just not set these headers if `has_data` is false rather than clear them later.